Rewrite example to use bmi abstract base class

mcflugen opened the pull request:
This pull request updates the *bmi-heat* example to use the `Bmi` abstract base class from the *bmipy* package. This mean removing a lot of unneeded code, including the `bmi` folder. In addition, the documentation is limited to this example. More general BMI documentation can be found elsewhere (i.e. [BMI](https://bmi.rtfd.io) and [BMI for Python](https://bmi-python.rtfd.io)).

mdpiper commented:
What about removing the `.bmi` folder, as well? We should cover how to componentize a model elsewhere.

mdpiper commented:
Also maybe remove the `.gitmodules` file? And does Landscape even work any more?

mcflugen commented:
@mdpiper good catch.

* removed the `.bmi` folder
* removed `.gitmodules`
* removed `.landscape.yaml`. I think landscape might still work but I switched to codacy, anyway.

Please have a look and see if everything looks good to you.
